com.github.sdbg.integration.jdt/src/com/github/sdbg/integration/jdt/JDTBreakpointPathResolver.java

```diff
@@ -33,6 +40,19 @@ public String getPath(IBreakpoint breakpoint) throws CoreException {
     }
   }
 
+  public String getLocalPath(IJavaBreakpoint bp) throws CoreException {
+    if (bp.getMarker() != null 
```

Comment:
Following notes:
- The code below will only work for breakpoints which are put inside .java files that are part of the Java project. It will not work for breakpoints that you put on files that come via the JRE container, or via the other library containers. 
- Is it really unavoidable to build up the file path by getting the directory path from the package name and then using the file name as-is by calling bp.getMarker().getResource().getName() ?

Reply:
- Problem is Java class name may not match file name. In case, if such class defined as a part of project that change would resolve filename correctly. For libraries, it would be as broken as before the change.
- Resource path is relative to project root, but we need path  relative to source root.

Reply:
OK makes sense. Only this is what I don't get: 
{quote}
Problem is Java class name may not match file name. In case, if such class defined as a part of project that change would resolve filename correctly. For libraries, it would be as broken as before the change.
{quote}

How come? Concrete example?

Reply:
This is jsweet example https://github.com/cincheo/jsweet-examples/blob/master/src/main/java/org/jsweet/examples/todomvc/ToDos.java
This java file contains a bunch of classes none of them is named ToDos.
Without fix breakpoints are not working.